content/en/user-guide/tools/transparent-endpoint-injection/dns-server.md

```diff
@@ -8,52 +8,88 @@ aliases:
+
 If you experience problems when running LocalStack and the DNS server is the issue, you can disable the DNS server using:
 
 ```bash
 DNS_ADDRESS=0
 ```
 
 {{< alert title="Warning" color="warning" >}}
-We do not recommend this configuration since this disables resolving `localhost.localstack.cloud` to the LocalStack container.
+We do not recommend disabling the DNS server since this disables resolving `localhost.localstack.cloud` to the LocalStack container.
 {{< / alert >}}
 
-You can also specify which exact URLs should be redirected to LocalStack by defining a hostname regex like:
+This option is primarily used by [LocalStack developers]({{< ref "contributing/development-environment-setup" >}}) in host mode because binding port 53 requires root privileges and port 53 might be occupied.
+
+### Fallback DNS server
+
+If you want to use another upstream DNS resolver than Google (default `8.8.8.8`),
+specify the fallback DNS server where all non-redirected queries (i.e., not matching `DNS_LOCAL_NAME_PATTERNS`) will be forwarded to:
 
 ```bash
-DNS_LOCAL_NAME_PATTERNS='.*(ecr|lambda).*.amazonaws.com'
+DNS_SERVER=1.1.1.1
 ```
 
-Using this configuration, the LocalStack DNS server only redirects ECR and Lambda domains to LocalStack, and the rest will be resolved via `$DNS_SERVER`. This can be used for hybrid setups, where certain API calls (e.g., ECR, Lambda) target LocalStack, whereas other services will target real AWS.
-
-{{< alert title="Warning" color="warning">}}
-We generally do not recommend connecting to real AWS from within LocalStack, in fact you should avoid using real AWS credentials anywhere in your LocalStack apps. Use this configuration with caution.
-{{< /alert >}}
+### Custom redirects
 
-There is the possibility to manually set the DNS server all not-redirected queries will be forwarded to:
+If you want to resolve only certain AWS URLs to LocalStack,
+specify a comma-separated list of hostname regex patterns such as:
 
 ```bash
-DNS_SERVER=1.1.1.1
+DNS_LOCAL_NAME_PATTERNS='.*(ecr|lambda).*.amazonaws.com'
 ```
 
-Per default, LocalStack uses the Google DNS resolver at `8.8.8.8`.
+Using this configuration, the LocalStack DNS server only resolves ECR and Lambda domains to LocalStack, and the rest will be resolved via the `DNS_SERVER` (i.e., the real DNS entry by default).
+This can be used for hybrid setups, where certain API calls (e.g., ECR, Lambda) target LocalStack, whereas other services will target real AWS.
+The regex pattern follows Python flavored-regex and can be tested at [regex101.com](https://regex101.com/r/OzIsQa/1).
+
+[The regex101 link is maintained by Joel Scheuner (requires linking to GitHub or Google account).
+It redirects to the main page if the saved example would not work.]: #
+
+{{< alert title="Warning" color="warning">}}
+Use this configuration with caution because we generally do not recommend connecting to real AWS from within LocalStack. 
+{{< /alert >}}
+
 
 ## Self-signed certificates
```

Comment:
I think this step could remain with the TEI docs, since it is not specific to the DNS server but rather TEI.

Reply:
I think this option does indeed refer to `DNS_LOCAL_NAME_PATTERNS` because skipping LocalStack resolution enables connecting to real AWS 🤔 (at least in it's original use)

I agree that it also fits for disabling transparent endpoint injection.